### DISCUSSION TITLE
Asset in Digital Ocean Spaces CDN

### DISCUSSION THREAD

**Discussion by altezza04** [directus/directus#4616] [Archive]
I want to check if an asset is served from let set API.mydomain.com/assets/abcd and it's store in Digital Ocean Spaces with CDN enabled, will the asset serve via digital ocean CDN? 

As far as I know Directus will route the asset from Spaces server, then process it before it serve to client.

CDN.mydomain.com/abcd -> API.mydomain.com/assets/abcd -> to the client

Is it correct?

**Comment by WoLfulus** [comment]
I was thinking about this over the weekend @rijkvanzanten 

A friend of mine that is using Directus asked me about it this week: the server he allocated Directus has a really low bandwidth allocated, but the storage service is priced on demand. This falls into what we discussed a while ago regarding signed requests. 

It would be nice if we could set a storage mode where Directus makes a redirect on the asset endpoint to a signed url on the storage service. That way we can still generate on-demand assets and still leverage traffic from the CDN.

We can provide this through extensions too so we don't need to change core and make specific implementations to different services.

**Reply by rijkvanzanten** [reply]
Sounds like an unrelated feature request to the OP 🙂

**Comment by rijkvanzanten** [comment]
> As far as I know Directus will route the asset from Spaces server, then process it before it serve to client.
> CDN.mydomain.com/abcd -> API.mydomain.com/assets/abcd -> to the client
> Is it correct?

Yes and no. The difference is that Directus pulls it straight from the bucket, not through DO's CDN 👍🏻
